staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_discovery_controller.go

```diff
@@ -142,11 +152,40 @@ func (c *DiscoveryController) sync(version schema.GroupVersion) error {
 			StorageVersionHash: storageVersionHash,
 		})
 
+		var scope apidiscoveryv2beta1.ResourceScope
```

Comment:
why aren't these blocks gated off

Reply:
When the gate is off, the resourceManager that ends-up consuming these objects is a "noOpResourceManager", I think that's a good pattern.

Reply:
>When the gate is off, the resourceManager that ends-up consuming these objects is a "noOpResourceManager", I think that's a good pattern.

Using gates to prevent calling a new code path protects callers from accidental NPEs in logic or picking up unpleasant behavior when not using the call path you're considering.  For instance, consider a few lines down when this is called `c.resourceManager.RemoveGroup(version.Group)`, that looks ripe for an NPE if we miss a construction case.  The straight value object assignment without any dereferencing is slightly less concerning, but "safe by side-effect" locations need addressing.

Reply:
This got fixed up over the course of other refactors.